python/kserve/kserve/model.py

```diff
@@ -347,6 +347,7 @@ async def _http_predict(
         except Exception as exc:
             logger.error(
                 f"Could not send a request to predictor at url {predict_url} "
+                f"with headers {predict_headers} "
```

Comment:
No information other than `Content-Type`, `x-request-id` and `x-b3-traceid` exist in this header. This should be safe to log.

Reply:
I am not sure if it really is all the headers that are added, however, for the Id ones, it might be good to log a small portion of it, if it is big enough to split in smaller pieces.

Reply:
It is actually all the headers added. The UI may not show it, it is defined a few lines above. Nevertheless, this code can change. To be more future proof I pulled the request id from the headers and logged only that.